core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -323,11 +390,6 @@ public HttpResponse doConfigSubmit(StaplerRequest req, StaplerResponse rsp) thro
             throw new ServletException("No such cloud " + this.name);
         }
         Cloud result = cloud.reconfigure(req, req.getSubmittedForm());
-        String proposedName = result.name;
```

Comment:
don't we still need validation as right now if I try rename the cloud through the configure page I get the same error this pull request is trying to solve?

![image](https://github.com/jenkinsci/jenkins/assets/21194782/7503a94a-042a-449a-88a4-80e078bab3fb)

Reply:
You're right, added check back in.